src/sequence_batch_scheduler.cc

```diff
@@ -813,6 +906,44 @@ SequenceBatchScheduler::DelayScheduler(
   return false;
 }
 
+void
+SequenceBatchScheduler::StartBackgroundThreads()
+{
+  // Create a reaper thread that watches for idle sequences.
+  reaper_thread_exit_ = false;
+  reaper_thread_.reset(
+      new std::thread([this]() { ReaperThread(10 /* nice */); }));
+
+  // Create a clean-up thread that asynchronously erase removed resources.
+  clean_up_thread_exit_ = false;
+  clean_up_thread_.reset(
+      new std::thread([this]() { CleanUpThread(20 /* nice */); }));
+}
+
+void
+SequenceBatchScheduler::StopBackgroundThreads()
+{
+  // Exit the clean-up thread.
+  {
+    std::lock_guard<std::mutex> lock(mu_);
+    clean_up_thread_exit_ = true;
+  }
+  clean_up_cv_.notify_one();
+  if (clean_up_thread_ && clean_up_thread_->joinable()) {
+    clean_up_thread_->join();
+  }
+
+  // Exit the reaper thread.
+  {
+    std::lock_guard<std::mutex> lock(mu_);
+    reaper_thread_exit_ = true;
```

Comment:
`std::atomic<bool>` would be simpler here.

Reply:
Yes, updated on commit: [Use atomic bool for thread exit vars](https://github.com/triton-inference-server/core/pull/207/commits/d875d79519ccdd3d0877161dea58ea4a02b503ea)